python/Ganga/test/GPI/FileTests/TestLocalFileClient.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+from Ganga.testlib.GangaUnitTest import GangaUnitTest
+from Ganga.testlib.file_utils import generate_unique_temp_file
+
+import datetime
+import time
+import os
+import shutil
+import copy
+
+class TestLocalFileClient(GangaUnitTest):
+    """test for sjid in filename names explain each test"""
+
+    _managed_files = []
+
+    # Num of sj in tests
+    sj_len = 3
+
+    # This sets up a LocalFileConfiguration which works by placing a file on local storage somewhere we can test using standard tools
+    LocalFileConfig = {'fileExtensions': [''],
+                       'uploadOptions': {},
+                       'backendPostprocess': {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client',
+                                              'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}}
+
+    def setUp(self):
+        """
+        Configure the LocalFile for the test
+        """
+        extra_opts=[('PollThread', 'autostart', 'False'),
+                    ('Local', 'remove_workdir', 'False'),
+                    ('TestingFramework', 'AutoCleanup', 'False'),
+                    ('Output', 'LocalFile', self.LocalFileConfig),
+                    ('Output', 'FailJobIfNoOutputMatched', 'True')]
+        super(TestLocalFileClient, self).setUp(extra_opts=extra_opts)
+
+    @staticmethod
+    def cleanUp():
+        """ Cleanup the current temp objects """
+
+        from Ganga.GPI import jobs
+        for j in jobs:
+            shutil.rmtree(j.backend.workdir, ignore_errors=True)
+            j.remove()
+
+        for file_ in TestLocalFileClient._managed_files:
+            os.unlink(file_)
+        TestLocalFileClient._managed_files = []
+
+    def test_a_testClientSideSubmit(self):
+        """Test the client side code whilst stil using the Local backend"""
+
+        from Ganga.GPI import LocalFile, Job, ArgSplitter
+
+        TestLocalFileClient.cleanUp()
```

Comment:
Are you sure? When the test starts jobs `jobs` and `_managed_files` will be `[]`. Therefore `for j in jobs:` will be a no-op, as will `for file_ in TestLocalFileClient._managed_files:`? The correct way to do this is to either use `unittest2`'s setup/teardown functions or a pytest fixture. Otherwise, writing a simple context manager to remove files when they're done with would work.